packages/aws-cdk-lib/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -153,6 +156,20 @@ export interface StateMachineProps {
    * @default RemovalPolicy.DESTROY
    */
   readonly removalPolicy?: RemovalPolicy;
+
+  /**
+   * KMS key used for encryption. It can either be a key or alias construct. If provided it will enable encryption.
+   * By default data is encrypted with an AWS owned key.
+   * @default - no kmsKeyId is associated
+   */
+  readonly kmsKey?: kms.IKey;
+
+  /**
+   * Maximum duration for which SFN will reuse data keys. When the period expires,
+   * Step Functions will call GenerateDataKey. This setting only applies to a customer managed key and does not apply to an AWS owned KMS key.
+   * @default - 300s
+   */
```

Comment:
* SFN -> Step Functions (be consistent)
* Formatting [suggested previously](https://github.com/aws/aws-cdk/pull/30959#discussion_r1701863772)
* Clarify it cannot be used unless `kmsKey` is provided (assuming we'll invalidate this)
* Clarify the value restrictions
* @default w/ JavaScript value

```suggestion
  /**
   * Maximum duration that Step Functions will reuse customer managed data keys.
   * When the period expires, Step Functions will call GenerateDataKey.
   * 
   * You can only provide a value if `kmsKey` is set.
   *
   * Must be between 60 and 900 seconds.
   *
   * @default Duration.seconds(300)
   */
```